src/rgw/rgw_rados.cc

```diff
@@ -7371,6 +7371,9 @@ int RGWRados::apply_olh_log(const DoutPrefixProvider *dpp,
   }
 
   r = bucket_index_trim_olh_log(dpp, bucket_info, state, obj->get_obj(), last_ver);
+  if (r == -ECANCELED) {
+    r = 0;
+  }
```

Comment:
if the olh tag in the bucket index changed, we should expect any other requests with `state->olh_tag` to fail the same way. returning success here would just cause `update_olh()` to loop again and make another call to `bucket_index_read_olh_log()` with this stale olh tag

we should return `ECANCELED` as-is and let the callers of `update_olh()` handle it

Reply:
Yeah I think that's fair. Should we go one step further and immediately return ECANCELED above if it's returned  by `rgw_rados_operate`, instead of ignoring it there?

Reply:
that sounds sensible. let's try it out and make sure it doesn't cause any test failures